### ISSUE TITLE
Fix join/leave messages

### ISSUE THREAD

**Issue by BadgerCode** [BadgerCode/emojichat#3]
The default ones don't seem to work.
```lua
hook.Add( "ChatText", "echat_joinleave", function( index, name, text, type )
	print("LOOK HERE IDIOT: [" .. type .. "]: " .. text)
end)
```

* Hook into player connect/initial spawn & disconnect.
* Make it so server owners can turn it off
* Maybe utilise cl_chatfilters

**cl_chatfilters**
https://github.com/ValveSoftware/Source-1-Games/issues/2245

> cl_chatfilters "63" // def. "63" Stores the chat filter settings 63 = 00111111 binary = all on.
(have it in here in case I forget that I turned off stuff)
> // 00000001 = 1, Show Joins/leaves in chat
> // 00000010 = 2, Show Name Changes in chat
> // 00000100 = 4, Show Public Chat (disable for only team chat ?? )
> // 00001000 = 8, Show Server Messages (if only this would work on messages from plugins...)
> // 00010000 = 16, Show Team Changes
> // 00100000 = 32, Show Achievement Announce
> // 00011111 = 31, Show: Joins/Leaves, NameChanges, Public chat, team changes.

**Comment by BadgerCode** [comment]
https://github.com/Facepunch/garrysmod-issues/issues/3565

**Comment by BadgerCode** [comment]
Fixed in #18
